SFS_general/delayline.m

```diff
@@ -95,112 +98,114 @@
 % a usage of fractional delay due to the upsampling.
 %
 switch delay.resampling
-    case 'none'
-        rfactor = 1.0;
-        delay_offset = 0.0;
-    case 'matlab'
-        rfactor = delay.resamplingfactor;
-        delay_offset = 0.0;
-        sig = resample(sig,rfactor,1);
-    case 'pm'
-        % === Parks-McClellan linear phase FIR filter ===
-        rfactor = delay.resamplingfactor;
-        delay_offset = delay.resamplingorder*rfactor / 2;
-        A = [1 1 0 0];
-        f = [0.0 0.9/rfactor 1/rfactor 1.0];
-        rfilt = rfactor*firpm(delay.resamplingorder*rfactor,f,A);
-
-        sig = reshape(sig,1,channels*samples);
-        sig = [sig; zeros(rfactor-1,channels*samples)];
-        sig = reshape(sig,rfactor*samples,channels);
-
-        sig = filter(rfilt,1,sig,[],1);
-    otherwise
-        error('%s: "%s": unknown resampling method',upper(mfilename), ...
-            delay.resampling);
+case 'none'
+    rfactor = 1.0;
+    delay_offset = 0.0;
+case 'matlab'
+    rfactor = delay.resamplingfactor;
+    delay_offset = 0.0;
+    sig = resample(sig,rfactor,1);
+case 'pm'
+    % === Parks-McClellan linear phase FIR filter ===
+    rfactor = delay.resamplingfactor;
+    delay_offset = (delay.resamplingorder*rfactor/2) / fs;
```

Comment:
I guess this is better as we avoid lots of `/ fs`. Originally I thought it might be a good idea to use always time, but `dt` is also converted to samples as you pointed out with the next comment.
So, I changed the behavior of `delay_offset` which is now converted into time only at the end of the function.